File: maximizeIt.py
from __future__ import print_function

def swap(ls):
    temp = ls[0]
    ls[0] = ls[1]
    ls[1] = temp
    return ls

n, k = int(input()), int(input())
ls = list()
for i in range(n):
	temp = list(map(int, input().split()))
	temp = swap(temp)
	ls.append(temp)
ls.sort()
ans = 0
arr = list()
for i in range(n):
	if ans + ls[i][0] <= k:
		arr.append(ls[i][1])
		ans += ls[i][0]
	else:
		break
if len(arr) > 0:
	arr.sort()
	for i in range(len(arr)):
		print(arr[i], end = "")
		if i < len(arr)-1:
			print(",", end = "")
else:
	print(-1)

File: maximizeIt.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main(void) {
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);
    
    long n, k;
    cin >> n >> k;
    vector<pair<long, long>> arr(n);
    for (int i = 0; i < n; i++)
        cin >> arr[i].second >> arr[i].first;
    sort(arr.begin(), arr.end());
    long ans = 0;
    vector<long> v;
    for (int i = 0; i < n; i++) {
        if (ans + arr[i].first <= k) {
            v.push_back(arr[i].second);
            ans += arr[i].first;
        } else {
            break;
        }
    }
    if (v.size() > 0) {
        sort(v.begin(), v.end());
        for (int i = 0; i < v.size(); i++) {
            cout << v[i];
            if (i < v.size()-1)
                cout << ",";
        }
    } else {
        cout << -1;
    }
    cout << endl;
}
